src/main/java/com/exasol/bucketfs/uploadnecassity/ChecksumUploadNecessityCheckStrategy.java

```diff
@@ -0,0 +1,144 @@
+package com.exasol.bucketfs.uploadnecassity;
+
+import static com.exasol.bucketfs.uploadnecassity.ByteArrayToHexConverter.toHex;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.*;
+import java.sql.*;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import com.exasol.bucketfs.*;
+import com.exasol.errorreporting.ExaError;
+
+/**
+ * This {@link UploadNecessityCheckStrategy} compares the checksum of the local file and the file on BucketFs and only
```

Comment:
This is a nice example of the "comments lie" observation from the "Clean Code" book, because the comment does not mention the fact that a size limit also plays a role. :-)